### ISSUE TITLE
Suggestion: when iterating array, warn about using the variable as index in that array

### ISSUE THREAD

**Issue by koalaman** [koalaman/shellcheck#881]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
for i in "${array[@]}"
do
  echo "${array[i]}"
done
```

This would be correct in e.g. Javascript or AWK, but not in Bash.

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

"Warnings: For loops iterate over array values, not over indices. "
